Automattic/jetpack
# CRM: WooSync keeps changing status

This PR makes sure WooSync only changes contact's statuses if it is the first one from the status list (CRM ships it with 'Lead' being the default value).

## Proposed changes:

* WooSync now changes a contact's status if it is equal to the first one from the list (default value is 'Lead')

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* You will need to have WooCommerce installed
* Map contact statuses to different values in the WooSync Settings page (`wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=woosync`) - It is important to have different statuses mapped to make sure they weren't changed later
* Create a Contact with an e-mail in `wp-admin/admin.php?page=zbs-add-edit&action=edit&zbstype=contact` (even better to enable it in Client Portal after adding)
* Make sure this contact's status is 'Lead'
* Add a WooCommerce order for this Contact, this order should have a Status that is mapped in the CRM to something different from 'Lead'
* Check if the status has changed (e.g. to the Customer status)
* Add another WooCommerce order for the same contact mapped to something different (e.g. 'Refused')

In `trunk`, after adding the second Woo order, the contact status changes (in that example to 'Refused')
In `fix/crm-woosync-keeps-changing-status`  after adding, the second Woo order the contact status remains the same (in that example 'Customer')
